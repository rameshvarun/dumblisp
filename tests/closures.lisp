(defun create-counter ()
  (print "Creating new counter...")
  (let ((i 0)) ; Start counter at 0
    (lambda ()
      (set i (+ i 1))
      i
      )))

; Create a counter
(let ((a (create-counter)))
  (print (a))
  (print (a))
  (print (a)))

; Create a second counter
(let ((a (create-counter)))
  (print (a))
  (print (a))
  (print (a)))
