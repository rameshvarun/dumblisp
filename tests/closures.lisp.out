Creating new counter...
1
2
3
Creating new counter...
1
2
3
